scripts/fuzzy-match-release-branch.sh

```diff
@@ -0,0 +1,177 @@
+  subject=$(git log -1 --pretty=format:"%s" "$commit")
+  meta_key="${subject}__${author}"
+  patch=$(git show --pretty=format:%b "$commit" | tail -n +2 | sed 's/^[[:space:]]*//')
+  patch_hash=$(echo "$patch" | hash_patch)
+
+  SRC_COMMIT_META+=("$meta_key")
+  SRC_PATCH_HASHES+=("$patch_hash")
+  SRC_PATCHES+=("$patch")
+done
+
+TOTAL=${#RELEASE_COMMITS_ARRAY[@]}
+MATCHED=0
+UNMATCHED=0
+
+for i in "${!RELEASE_COMMITS_ARRAY[@]}"; do
+  rc_commit="${RELEASE_COMMITS_ARRAY[$i]}"
+  rc_author=$(git log -1 --pretty=format:"%an <%ae>" "$rc_commit")
+  rc_subject=$(git log -1 --pretty=format:"%s" "$rc_commit")
+  meta_key="${rc_subject}__${rc_author}"
+
+  echo -ne "[$((i + 1))/$TOTAL] Checking ${rc_commit:0:7}... "
+
+  rc_patch=$(git show --pretty=format:%b "$rc_commit" | tail -n +2 | sed 's/^[[:space:]]*//')
+  rc_patch_hash=$(echo "$rc_patch" | hash_patch)
+
+  found_exact=""
+  for j in "${!SRC_PATCH_HASHES[@]}"; do
+    if [[ "${SRC_PATCH_HASHES[$j]}" == "$rc_patch_hash" ]]; then
+      found_exact="${SRC_COMMITS_ARRAY[$j]}"
+      break
+    fi
+  done
+
+  if [[ -n "$found_exact" ]]; then
+    src_subject=$(git log -1 --pretty=format:"%s" "$found_exact")
+    src_author=$(git log -1 --pretty=format:"%an <%ae>" "$found_exact")
+    echo "✅ MATCHES ${found_exact:0:7}"
+    echo "    ↪ RELEASE: $rc_commit  $rc_author"
+    echo "              \"$rc_subject\""
+    echo "    ↪ SOURCE : $found_exact  $src_author"
+    echo "              \"$src_subject\""
+    echo ""
+    MATCHED=$((MATCHED + 1))
+    continue
+  fi
+
+  echo "❌ NO MATCH"
+  UNMATCHED=$((UNMATCHED + 1))
+
+  echo "🔍 Unmatched Commit:"
+  echo "    ↪ Commit : $rc_commit"
+  echo "    ↪ Author : $rc_author"
+  echo "    ↪ Subject: \"$rc_subject\""
+  echo ""
+
+  # Try fuzzy match only within commits that share author+subject
+  best_score=99999
+  best_index=""
+  for j in "${!SRC_COMMIT_META[@]}"; do
+    if [[ "${SRC_COMMIT_META[$j]}" == "$meta_key" ]]; then
+      diff=$(diff -u <(echo "$rc_patch") <(echo "${SRC_PATCHES[$j]}") || true)
+      score=$(echo "$diff" | grep '^[-+]' | wc -l | xargs)
+      if [[ "$score" -lt "$best_score" ]]; then
+        best_score=$score
+        best_index=$j
+      fi
+    fi
+  done
+
+  if [[ -n "$best_index" ]]; then
+    match_commit="${SRC_COMMITS_ARRAY[$best_index]}"
+    match_author=$(git log -1 --pretty=format:"%an <%ae>" "$match_commit")
+    match_subject=$(git log -1 --pretty=format:"%s" "$match_commit")
+
+    echo "🤔 Closest fuzzy match: $match_commit  ($best_score changed lines)"
+    echo "    ↪ Author : $match_author"
+    echo "    ↪ Subject: \"$match_subject\""
+    echo ""
+    echo "🔧 Check it manually:"
+    echo "    git diff --histogram $rc_commit $match_commit"
+  else
+    echo "⚠️ No commits with matching author + subject in source branch."
+  fi
+
+  echo ""
+  echo "🔍 Diff of release commit:"
+  echo "---------------------------------------------"
+  git show "$rc_commit" | sed 's/^/    /'
```

Comment:
Instead of showing the change of the diff, could we show the difference of the release commit and the closest fuzzy match commit?

Reply:
Done.